src/test/es6-module-bundling-scenarios_test.ts

```diff
@@ -24,6 +24,45 @@ import {heredoc, inMemoryAnalyzer} from './test-utils';
 
 suite('Es6 Module Bundling', () => {
 
+  test('export from', async () => {
+    const analyzer = inMemoryAnalyzer({
+      'a.js': `
+        export * from './b.js';
+        export const A = 'a';
+      `,
+      'b.js': `
+        export * from './c.js';
+        export const B = 'b';
+      `,
+      'c.js': `
+        export const C = 'c';
+        export {C as default};
+      `,
+    });
+    const aUrl = analyzer.resolveUrl('a.js')!;
+    const bundler = new Bundler({analyzer});
+    const {documents} =
+        await bundler.bundle(await bundler.generateManifest([aUrl]));
+    assert.deepEqual(documents.get(aUrl)!.content, heredoc`
+      const C = 'c';
+      var c = {
+        C: C,
+        default: C
+      };
+      const B = 'b';
+      var b = {
+        B: B,
+        C: C
+      };
+      const A = 'a';
+      var a = {
+        A: A,
+        B: B,
+        C: C
+      };
+      export { a as $a, b as $b, c as $c, C, B, A, C as C$1, B as B$1, C as C$2, C as $cDefault };`);
```

Comment:
If this is a representation of `a.js` then shouldn't we _not_ be exporting `$cDefault`?

Reply:
This is a simplifying bundler behavior that all modules that are part of a bundle export their exports from the bundle under a namespace for their module.

It's because we don't currently track which modules are depended on outside the bundle directly.  This is an optimization we may choose to make.  Filed: https://github.com/Polymer/polymer-bundler/issues/650